src/components/PostList.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import { Post } from './Post';
+import { SearchForm } from './SearchForm';
+import useDebounce from './UseDebounce';
+
+type PostListProps = {
+  posts: PreparedPost[];
+};
+
+const PostList: React.FC<PostListProps> = ({ posts }) => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [suitablePosts, setSuitablePosts] = useState<PreparedPost[]>(posts);
+
+  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setInputValue(value);
+  };
+
+  const searchMatchedPosts = (value: string) => {
+    const regexp = new RegExp(value, 'i');
+    const matchedPosts = posts.filter(post => (regexp.test(post.title) || regexp.test(post.body)));
+
+    return matchedPosts;
+  };
```

Comment:
```suggestion
  const searchMatchedPosts = (value: string) => {
    const regexp = new RegExp(value, 'i');
    const matchedPosts = posts.filter(post => (regexp.test(post.title) || regexp.test(post.body)));

    return matchedPosts;
  };
```
use useMemo to improve performance